[Bug] Creating an SQLJob with "unknown" dependency get the operator to crash

<!--
Bugs should be filed for issues encountered whilst operating mariadb-operator.
Please provide as much detail as possible. 
-->

**Documentation**
- [x ] I acknowledge that I have read the relevant [documentation](https://github.com/mariadb-operator/mariadb-operator/tree/main/docs).

**Describe the bug**
<!--
If you create an SQLJob with mariadb operator ghcr.io/mariadb-operator/mariadb-operator:v0.0.27 you get it to crash with the following log:
```
{"level":"info","ts":1713790751.308197,"msg":"Dependency 'job-not-created' not found","controller":"sqljob","controllerGroup":"k8s.mariadb.com","controllerKind":"SqlJob","SqlJob":{"name":"with-with-wrong-dependency-058a94b62a","namespace":"bridge"},"namespace":"bridge","name":"with-with-wrong-dependency-058a94b62a","reconcileID":"e5375ccf-9a74-4e17-ad89-7fbec7b6baee"}
{"level":"info","ts":1713790751.3222082,"msg":"Observed a panic in reconciler: runtime error: invalid memory address or nil pointer dereference","controller":"sqljob","controllerGroup":"k8s.mariadb.com","controllerKind":"SqlJob","SqlJob":{"name":"with-with-wrong-dependency-058a94b62a","namespace":"bridge"},"namespace":"bridge","name":"with-with-wrong-dependency-058a94b62a","reconcileID":"e5375ccf-9a74-4e17-ad89-7fbec7b6baee"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x348 pc=0x1712c46]

goroutine 579 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.2/pkg/internal/controller/controller.go:116 +0x1e5
panic({0x193a800?, 0x2cb8ac0?})
	/usr/local/go/src/runtime/panic.go:914 +0x21f
github.com/mariadb-operator/mariadb-operator/api/v1alpha1.(*SqlJob).IsComplete(...)
	/app/api/v1alpha1/sqljob_types.go:103
github.com/mariadb-operator/mariadb-operator/controller.(*SqlJobReconciler).waitForDependencies(0xc0003f4d70, {0x1e895f0, 0xc000d95cb0}, 0xc000143180)
	/app/controller/sqljob_controller.go:132 +0x2e6
github.com/mariadb-operator/mariadb-operator/controller.(*SqlJobReconciler).Reconcile(0xc0003f4d70, {0x1e895f0, 0xc000d95cb0}, {{{0xc000afd826?, 0x5?}, {0xc000afeea0?, 0xc000cc5d48?}}})
	/app/controller/sqljob_controller.go:61 +0x1c5
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x1e8d788?, {0x1e895f0?, 0xc000d95cb0?}, {{{0xc000afd826?, 0xb?}, {0xc000afeea0?, 0x0?}}})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.2/pkg/internal/controller/controller.go:119 +0xb7
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000122aa0, {0x1e89628, 0xc000416ff0}, {0x19cddc0?, 0xc000c57b00?})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.2/pkg/internal/controller/controller.go:316 +0x3cc
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000122aa0, {0x1e89628, 0xc000416ff0})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.2/pkg/internal/controller/controller.go:266 +0x1c9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.2/pkg/internal/controller/controller.go:227 +0x79
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2 in goroutine 132
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.2/pkg/internal/controller/controller.go:223 +0x565```
for code blocks of your kubectl output or YAML files.
-->

**Expected behaviour**
<!--A concise description of what you expected to happen.-->
Not starting the SqlJob before the dependency exists and are OK. Not crashing

**Steps to reproduce the bug**
<!--Steps to reproduce the bug should be clear and easily reproducible to help people
gain an understanding of the problem.-->
Create a job like:

```
apiVersion: k8s.mariadb.com/v1alpha1
kind: SqlJob
metadata:
  name: my-name
spec:
  restartPolicy: Never
  dependsOn:
    - name: not-existing-job
  mariaDbRef:
    name: mariadb
  username: root
  passwordSecretKeyRef:
    name: mariadb-root
    key: password
  database: my-db
  sql: "use db"
```

**Debug information**
I have added logs above, please let me know if you want anything more.

**Environment details**:
- Kubernetes version: 1.28.3
- mariadb-operator version: 0.0.27
- Install method: helm+argocd

**Additional context**
<!--Add any other context  here.-->

--- a/internal/controller/sqljob_controller.go
+++ b/internal/controller/sqljob_controller.go
@@ -125,9 +125,7 @@ func (r *SqlJobReconciler) waitForDependencies(ctx context.Context, sqlJob *v1al
 			}
 
 			logger.Info(msg)
-			if err := r.patchStatus(ctx, sqlJob, r.ConditionComplete.PatcherFailed(msg)); err != nil {
-				return false, ctrl.Result{}, err
-			}
+			return false, ctrl.Result{RequeueAfter: r.RequeueInterval}, r.patchStatus(ctx, sqlJob, r.ConditionComplete.PatcherFailed(msg))
 		}
 		if !sqlJobDep.IsComplete() {
 			msg := fmt.Sprintf("Dependency '%s' not ready", dep.Name)

--- a/internal/controller/sqljob_controller_test.go
+++ b/internal/controller/sqljob_controller_test.go
@@ -2,12 +2,14 @@ package controller
 
 import (
 	"reflect"
+	"time"
 
 	mariadbv1alpha1 "github.com/mariadb-operator/mariadb-operator/api/v1alpha1"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 	batchv1 "k8s.io/api/batch/v1"
 	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/utils/ptr"
@@ -197,6 +199,59 @@ var _ = Describe("SqlJob", func() {
 			Expect(svcAcc.ObjectMeta.Annotations).NotTo(BeNil())
 			Expect(svcAcc.ObjectMeta.Annotations).To(HaveKeyWithValue("k8s.mariadb.com/test", "test"))
 		}
+	})
+
+	It("should report unready status for non-existent dependencies", func() {
+		nonExistentDependency := "not-existing-job"
+		sqlJob := mariadbv1alpha1.SqlJob{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "sqljob-with-non-existent-dependency",
+				Namespace: testNamespace,
+			},
+			Spec: mariadbv1alpha1.SqlJobSpec{
+				DependsOn: []mariadbv1alpha1.LocalObjectReference{
+					{
+						Name: nonExistentDependency,
+					},
+				},
+				MariaDBRef: mariadbv1alpha1.MariaDBRef{
+					ObjectReference: mariadbv1alpha1.ObjectReference{
+						Name: testMdbkey.Name,
+					},
+					WaitForIt: true,
+				},
+				Username: testUser,
+				PasswordSecretKeyRef: mariadbv1alpha1.SecretKeySelector{
+					LocalObjectReference: mariadbv1alpha1.LocalObjectReference{
+						Name: testPwdKey.Name,
+					},
+					Key: testPwdSecretKey,
+				},
+				Database: &testDatabase,
+			},
+		}
+
+		By("Creating SqlJob with non-existent dependency")
+		Expect(k8sClient.Create(testCtx, &sqlJob)).To(Succeed())
+		DeferCleanup(func() {
+			Expect(k8sClient.Delete(testCtx, &sqlJob)).To(Succeed())
+		})
+
+		By("Expecting SqlJob to report unready status")
+		Consistently(func() bool {
+			var updatedSqlJob mariadbv1alpha1.SqlJob
+			if err := k8sClient.Get(testCtx, client.ObjectKeyFromObject(&sqlJob), &updatedSqlJob); err != nil {
+				return false
+			}
+			return !updatedSqlJob.IsComplete()
+		}, 10*time.Second, testInterval).Should(BeTrue())
+
+		By("Expecting SqlJob to not create a Job")
+		Consistently(func() bool {
+			var job batchv1.Job
+			err := k8sClient.Get(testCtx, client.ObjectKeyFromObject(&sqlJob), &job)
+			return err != nil && apierrors.IsNotFound(err)
+		}, 10*time.Second, testInterval).Should(BeTrue())
 	})
 
 	DescribeTable("Creating an SqlJob",

